File: app/iOS/Thermal Comfort Study/BottomClothing.swift
//
//  BottomClothing.swift
//  Thermal Comfort Study
//
//  Created by Madhav Achar on 10/20/16.
//  Copyright © 2016 boschsc. All rights reserved.
//

import Foundation

public enum BottomClothing {
    case SHORTS
    case LONG_PANTS_THIN
    case LONG_PANTS_THICK
    case SKIRT_THIN
    case SKIRT_THICK
    case NONE
    
    // Declaration for UIPicker
    static let all = [SHORTS, LONG_PANTS_THIN, LONG_PANTS_THICK, SKIRT_THIN,
                      SKIRT_THICK, NONE]
    
    // Return access to a tuple of values
    public func values() -> (insulation: Double, name: String, description: String) {
        switch self {
        case .SHORTS:
            return (0.08, "Shorts", "Walking shorts")
        case .LONG_PANTS_THIN:
            return (0.15, "Long pants (thin)", "Straight trousers (thin)")
        case .LONG_PANTS_THICK:
            return (0.24, "Long pants (thick)", "Straight trousers (thick)")
        case .SKIRT_THIN:
            return (0.14, "Skirt (thin)", "Skirt (thin)")
        case .SKIRT_THICK:
            return (0.23, "Skirt (thick)", "Skirt (thick)")
        case .NONE:
            return (0.00, "None", "None")
        }
    }
    
}

File: app/android/app/src/main/java/bosch/smartcampus/thermalcomfortstudy/lib/BottomClothing.java
package bosch.smartcampus.thermalcomfortstudy.lib;

/**
 * Created by rsukkerd on 6/14/16.
 */
public enum BottomClothing {
    D2 (0.08, "Shorts", "Walking shorts"),
    D3 (0.15, "Long pants (thin)", "Straight trousers (thin)"),
    D4 (0.24, "Long pants (thick)", "Straight trousers (thick)"),
    E1 (0.14, "Skirt (thin)", "Skirt (thin)"),
    E2 (0.23, "Skirt (thick)", "Skirt (thick)");

    private final double clothingInsulation;
    private final String name;
    private final String description;

    BottomClothing(double clothingInsulation, String name, String description) {
        this.clothingInsulation = clothingInsulation;
        this.name = name;
        this.description = description;
    }

    public double getClothingInsulation() {
        return clothingInsulation;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }
}
